src/main/java/org/wildfly/security/authz/PermissionMappable.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.authz;
+
+import java.security.Principal;
+import java.time.Instant;
+
+import org.wildfly.security.auth.principal.AnonymousPrincipal;
+
+/**
+ * An entity to which permissions can be mapped.
+ *
+ * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ */
+public interface PermissionMappable {
```

Comment:
Only one set of roles is active, but which one is active depends on the context.  In this case it just becomes one more ephemeral object to instantiate before calling the permission mapper.